publish-report/action.yml

```diff
@@ -0,0 +1,35 @@
+name: "Publish Playwright report"
+description: "Publish a Playwright report to GCS"
+
+inputs:
+  path:
+    default: "playwright-report"
+    required: false
+    description: "Path to the folder to publish. Defaults to 'playwright-report'"
+
+runs:
+  using: "composite"
+  steps:
+    - id: get-secrets
+      uses: grafana/shared-workflows/actions/get-vault-secrets@main
+      with:
+        common_secrets: |
+          GCP_UPLOAD_ARTIFACTS_KEY=grafana/artifacts-uploader-service-account:'credentials.json'
+
+    - name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v2'
+      with:
+        credentials_json: '${{ env.GCP_UPLOAD_ARTIFACTS_KEY }}'
+
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v2'
+
+    - name: Publish report to Cloud Storage
+      shell: bash
+      run: |
+        echo "Publish release to Google Cloud Storage:"
+        gsutil cp -r ${{ inputs.directory }}. gs://releng-pipeline-artifacts-dev/${{ github.event.repository.name }}/${{ github.event.number }}/Grafana-v${{ matrix.GRAFANA_VERSION }}
+
+    - name: Write report link to summary
+      shell: bash
```

Comment:
just curious, what does `shell: bash` do in this context / why is it necessary?

Reply:
To run commands in [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action#creating-an-action-metadata-file) you need to specify `shell`. Also see https://github.com/orgs/community/discussions/18597.